rmw_dds_common/src/security.cpp

```diff
@@ -80,68 +59,37 @@ bool get_security_files(
   const std::string & secure_root,
   std::unordered_map<std::string, std::string> & result)
 {
-  using std::placeholders::_1;
-  using std::placeholders::_2;
-  using std::placeholders::_3;
-  using std::placeholders::_4;
-  using security_file_processor =
-    std::function<bool (bool, const std::string &, const std::filesystem::path &, std::string &)>;
-  using processor_vector =
-    std::vector<std::pair<std::string, security_file_processor>>;
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  rcutils_string_map_t security_files = rcutils_get_zero_initialized_string_map();
+  rcutils_ret_t ret = rcutils_string_map_init(&security_files, 0, allocator);
 
-  // Key: the security attribute
-  // Value: ordered sequence of pairs. Each pair contains one possible file name
-  //        for the attribute and the corresponding processor method
-  // Pairs are ordered by priority: the first one matching is used.
-  const std::unordered_map<std::string, processor_vector> required_files{
-    {"IDENTITY_CA", {
-        {"identity_ca.cert.p11", std::bind(process_pkcs_uri_security_file, _1, _2, _3, _4)},
-        {"identity_ca.cert.pem", std::bind(process_file_uri_security_file, _1, _2, _3, _4)}}},
-    {"CERTIFICATE", {
-        {"cert.p11", std::bind(process_pkcs_uri_security_file, _1, _2, _3, _4)},
-        {"cert.pem", std::bind(process_file_uri_security_file, _1, _2, _3, _4)}}},
-    {"PRIVATE_KEY", {
-        {"key.p11", std::bind(process_pkcs_uri_security_file, _1, _2, _3, _4)},
-        {"key.pem", std::bind(process_file_uri_security_file, _1, _2, _3, _4)}}},
-    {"PERMISSIONS_CA", {
-        {"permissions_ca.cert.p11", std::bind(process_pkcs_uri_security_file, _1, _2, _3, _4)},
-        {"permissions_ca.cert.pem", std::bind(process_file_uri_security_file, _1, _2, _3, _4)}}},
-    {"GOVERNANCE", {
-        {"governance.p7s", std::bind(process_file_uri_security_file, _1, _2, _3, _4)}}},
-    {"PERMISSIONS", {
-        {"permissions.p7s", std::bind(process_file_uri_security_file, _1, _2, _3, _4)}}},
-  };
+  if (ret != RCUTILS_RET_OK) {
+    RMW_SET_ERROR_MSG("error initializin map");
+    return false;
```

Comment:
I think `rcutils_string_map_fini` needs to be called before return, and any other following error return cases.